R/utils.R

```diff
@@ -34,3 +34,23 @@ calc_weights <- function(net) {
     sfnetworks::activate("edges") |>
     dplyr::mutate(weight = sfnetworks::edge_length())
 }
+
+#' Get UTM zone from longitude
+#'
+#' @param x An sf object
+#'
+#' @return The UTM zone
+#' @export
+get_utm_zone_epsg <- function(x) {
+  if (!"sf" %in% class(x)) {
+    stop("x must be an sf object")
+  }
+
+  long <- x |>
+    sf::st_transform(4326) |>
+    sf::st_union() |>
+    sf::st_centroid() |>
+    sf::st_coordinates()
```

Comment:
Probably it does not matter too much, but if `x` contains many/complicated geometries, the reprojection/union/calculation of the centroid can become expensive. What about the following alternative approach, not exactly equivalent, but maybe good enough for the purpose?

Also I suggest to rename the variable to `coords`, see following comment.

```suggestion
  coords <- x |>
    sf::st_bbox() |>
    sf::st_as_sfc() |>
    sf::st_centroid() |>
    sf::st_transform(4326) |>
    sf::st_coordinates()
```